CMake/UpdateSubmodules.cmake

```diff
@@ -0,0 +1,14 @@
+find_package(Git)
+if (GIT_FOUND)
+	option(UPDATE_SUBMODULES "Download missing submodules during build" ON)
+	if (UPDATE_SUBMODULES )
+		message("Updating submodules.")
+		execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
```

Comment:
Yes I recall you doing this. I'll probably replace this with an immediate error so that it's more obvious to the user why it's failing (rather than throwing a compile error that a file is missing). Could you elaborate on what issues you ran into with this approach?

Reply:
I agree with @joaander - it gets clunky when changing submodules.

Reply:
As a developer, it is frustrating when you make a change to submodule, compile, and then that change is gone. It is also frustrating that the very slow `git submodule update --init` process occurs every time you compile. As a user who builds the code only once, it is a minor teaching moment to learn to use `git clone --recursive` or `git submodule update --init`.

Reply:
Makes sense. The current build script just raises errors to indicate when submodules are not present, which I think is a useful compromise.